pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -245,8 +267,7 @@ func (r *ReconcileOneAgent) reconcileRollout(logger logr.Logger, instance *dynat
 	if instance.Status.Version == "" {
 		desired, err := dtc.GetLatestAgentVersion(dtclient.OsUnix, dtclient.InstallerTypeDefault)
 		if err != nil {
-			logger.Error(err, "failed to get desired version")
-			return updateCR, nil
+			return updateCR, fmt.Errorf("failed to get desired version: %w", err)
```

Comment:
This is technically a change of semantics, reconcileVersion() wouldn't now run if failed to get the latest version from the API. I think it's fine however.

Reply:
Should we adapt it?

Reply:
Let's leave it for a follow-up refactor. Although to be fair this is probably better anyway.

Reply:
Will leave it then :)